htdocs/societe/card.php

```diff
@@ -2820,7 +2820,14 @@ function init_check_no_email(input) {
 				print '<tr>';
 				print '<td>'.$idprof.'</td><td>';
 				$key = 'idprof'.$i;
-				print dol_print_profids($object->$key, 'ProfId'.$i, $object->country_code, 1);
+
+				if ($i == 3) {
```

Comment:
Other countries may not use the . as separator for their profid3.
So i think forcing to . must be done into dol_print_profids for FR and i = 3 only

Reply:
Correct. That's what I intended first. But then I saw the `dol_print_profids()` signature allowing to chose the separator and thought "well, we'll use the `$separ` argument then, it's here for a reason". 

However I think it is misleading allowing deciding over the formatting in two different places.

Especially if this `$separ`argument is forced in some cases. 
For eg. one could call `dol_print_profids($object->$key, 'ProfId3', 'FR', 1, '-')` thinking this would replace the "." by a "-". But that wouldn't.

What is the benefit in keeping the `$separ` argument in the function definition?

Reply:
I don't know/remeber why we have the param separ. I see no reasin to have it. Logic should be inside the method. Did you analyse in current if this param was used by some callers ? If not we can remove it.

Reply:
In the develop branch, a `grep -rl "dol_print_profids("` returns 3 files having this method: 
- `htdocs/societe/card.php`
- `htdocs/main.inc.php`
- `htdocs/core/lib/functions.lib.php`

And none calls this method using the `$separ` parameter:
- `htdocs/societe/card.php` => 2 calls : 
https://github.com/Dolibarr/dolibarr/blob/787c838e43f49e105701416e18728fb5ebca87e6/htdocs/societe/card.php#L2823
https://github.com/Dolibarr/dolibarr/blob/787c838e43f49e105701416e18728fb5ebca87e6/htdocs/societe/card.php#L2938
- `htdocs/main.inc.php` => 7 calls:
https://github.com/Dolibarr/dolibarr/blob/787c838e43f49e105701416e18728fb5ebca87e6/htdocs/main.inc.php#L2175-L2193